Linear_cell_complex/include/CGAL/boost/graph/cgal_bgl_graph_io.h

```diff
@@ -0,0 +1,223 @@
+
+  typename boost::graph_traits<FaceGraph>::halfedge_descriptor
+    hdinit=halfedge(f, g), hd=hdinit;
+
+  do
+  {
+    ++res;
+    hd=next(hd, g);
+  }
+  while(hd!=hdinit);
+  
+  return res;
+}
+  
+/*!
+   \ingroup PkgBGLIOFct
+    writes the graph `g` in the OFF format.
+    \sa Overloads of this function for specific models of the concept `FaceGraph`.
+
+  */ 
+template <typename FaceGraph>
+bool write_off(std::ostream& os,
+               const FaceGraph& g)
+{
+  typedef typename boost::graph_traits<FaceGraph>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::graph_traits<FaceGraph>::face_descriptor face_descriptor;
+  typedef typename boost::graph_traits<FaceGraph>::vertices_size_type vertices_size_type;
+  typedef typename boost::graph_traits<FaceGraph>::faces_size_type faces_size_type;
+
+  typename boost::property_map<FaceGraph, CGAL::vertex_point_t>::const_type vpm = get(CGAL::vertex_point,g);
+  vertices_size_type nv = static_cast<vertices_size_type>(std::distance(vertices(g).first, vertices(g).second));
+  faces_size_type nf = static_cast<faces_size_type>(std::distance(faces(g).first, faces(g).second));
+
+  os << "OFF\n" << nv << " " << nf << " 0\n";
+  boost::container::flat_map<vertex_descriptor,vertices_size_type> reindex;
+  int n = 0;
+  BOOST_FOREACH(vertex_descriptor v, vertices(g)){
+    os << get(vpm,v) << '\n';
+    reindex[v]=n++;
+  }
+  
+  BOOST_FOREACH(face_descriptor f, faces(g)){
+    os << degree(f,g);
+    BOOST_FOREACH(vertex_descriptor v, vertices_around_face(halfedge(f,g),g)){
+      os << " " << reindex[v];
+    }
+    os << '\n';
+  }
+  return os.good();
+}
+
+
+/*!
+   \ingroup PkgBGLIOFct
+    writes the graph `g` in the OFF format.
+    \sa Overloads of this function for specific models of the concept `FaceGraph`.
+
+  */ 
+template <typename FaceGraph>
+bool write_off(const char* fname,
+               const FaceGraph& g)
+{
+  std::ofstream out(fname);
+  if(out.good()){
+    return write_off(out,g);
+  }
+  return false;
+}
+
+
+  namespace internal { namespace read_off {
+  
+  bool is_whitespace(const std::string& s)
+  {
+    for(int i=0; i < s.size(); i++){
+      if(s[i] != ' ' && s[i] != '\t'){
+        return false;
+      }
+    }
+    return true;
+  }
+  
+std::string next_non_comment(std::istream& is)
+{
+  std::string line;
+  do {
+    std::getline(is, line);
+  }while(line[0] == '#' || is_whitespace(line));
+  return line;
+}
+
+    }
+  } // namespace internal
+
+
+/*!
+   \ingroup PkgBGLIOFct
+    reads the graph `g` from data in the OFF format. Ignores comment lines which start with a hash, and lines with whitespace.
+    \sa Overloads of this function for specific models of the concept `FaceGraph`.
+    \pre The data must represent a 2-manifold
+    \attention The graph `g` is not cleared, and the data from the stream are added.
+
+  */ 
+template <typename FaceGraph>
+bool read_off(std::istream& is,
+              FaceGraph& g)
+{
```

Comment:
Yes this is a copy of the version of Andreas. 
This is of course temporary and will be removed when the small feature will be integrated.